image_services/egis/make_egis_services.py

```diff
@@ -0,0 +1,329 @@
+SEASONS = {
+    'summer': {
+        'SeasonCode': 'JJA',
+        'Season': 'June/July/August',
+        'StartDate': '06/01/2020',
+        'EndDate': '08/31/2020',
+    },
+    'fall': {
+        'SeasonCode': 'SON',
+        'Season': 'September/October/November',
+        'StartDate': '09/01/2020',
+        'EndDate': '11/30/2020',
+    },
+    'winter': {
+        'SeasonCode': 'DJF',
+        'Season': 'December/January/February',
+        'StartDate': '12/01/2019',
+        'EndDate': '02/29/2020',
+    },
+    'spring': {
+        'SeasonCode': 'MAM',
+        'Season': 'March/April/May',
+        'StartDate': '03/01/2020',
+        'EndDate': '05/31/2020',
+    }
+}
+
+
+def get_rasters(bucket: str, prefix: str, suffix: str) -> List[str]:
+    rasters = []
+    s3 = boto3.client('s3')
+    paginator = s3.get_paginator('list_objects_v2')
+    for page in paginator.paginate(Bucket=bucket, Prefix=prefix):
+        for obj in page['Contents']:
+            if obj['Key'].endswith(suffix):
+                rasters.append(f'/vsis3/{bucket}/{obj["Key"]}')
+    return rasters
+
+
+def get_pixel_type(data_type: str) -> int:
+    if data_type == 'Byte':
+        return 3
+    if data_type == 'Float32':
+        return 9
+    raise ValueError(f'Unsupported data type: {data_type}')
+
+
+def get_projection(srs_wkt: str) -> str:
+    srs = osr.SpatialReference()
+    srs.ImportFromWkt(srs_wkt)
+    return srs.GetAttrValue('AUTHORITY', 1)
+
+
+def get_raster_metadata(raster_path: str) -> dict:
+    assert raster_path.startswith('/vsis3/sentinel-1-global-coherence-earthbigdata/')
+    key = raster_path.removeprefix('/vsis3/sentinel-1-global-coherence-earthbigdata/')
+    download_url = f'https://sentinel-1-global-coherence-earthbigdata.s3.us-west-2.amazonaws.com/{key}'
+
+    name = Path(raster_path).stem
+    tile, season, polarization, product_type = name.split('_')
+    polarization = polarization.upper()
+    tag = f'{product_type}_{polarization}_{SEASONS[season]["SeasonCode"]}'
+
+    info = gdal.Info(raster_path, format='json')
+    return {
+        'Raster': raster_path,
+        'Name': name,
+        'xMin': info['cornerCoordinates']['lowerLeft'][0],
+        'yMin': info['cornerCoordinates']['lowerLeft'][1],
+        'xMax': info['cornerCoordinates']['upperRight'][0],
+        'yMax': info['cornerCoordinates']['upperRight'][1],
+        'nRows': info['size'][1],
+        'nCols': info['size'][0],
+        'nBands': len(info['bands']),
+        'PixelType': get_pixel_type(info['bands'][0]['type']),
+        'SRS': get_projection(info['coordinateSystem']['wkt']),
+        'Tag': tag,
+        'GroupName': tag,
+        'StartDate': SEASONS[season]['StartDate'],
+        'EndDate': SEASONS[season]['EndDate'],
+        'ProductType': product_type,
+        'Season': SEASONS[season]['Season'],
+        'Polarization': polarization,
+        'Tile': tile,
+        'DownloadURL': download_url,
+        'URLDisplay': name,
+        'MaxPS': '910',
```

Comment:
Unfortunately, the `MaxPS` here doesn't seem to be respected. Running the current code gives me `MaxPS=927.66242`. Probably best to bring the `CalculateFields` call back just to set `MaxPS` (reverting https://github.com/ASFHyP3/gis-services/pull/116/commits/26c2f066621e7b134417e89a6aafe5e817e76b4b)